checker.c

```diff
@@ -1,21 +1,48 @@
-#include <stdio.h>
-#include <assert.h>
+#include "checker.h"
 
-int batteryIsOk(float temperature, float soc, float chargeRate) {
-  if(temperature < 0 || temperature > 45) {
-    printf("Temperature out of range!\n");
-    return 0;
-  } else if(soc < 20 || soc > 80) {
-    printf("State of Charge out of range!\n");
-    return 0;
-  } else if(chargeRate > 0.8) {
-    printf("Charge Rate out of range!\n");
-    return 0;
-  }
-  return 1;
+float readtemperature(float temperature)
+{
+    if((temperature < TEMP_MIN) || (temperature > TEMP_MAX))
+    {
+        printf("Temperature out of range!\n");
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+float readsoc(float soc)
+{
+    if(soc < SOC_MIN || soc > SOC_MAX)
+    {
+        printf("State of Charge out of range!\n");
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+float readchargeRate(float chargeRate)
+{
+    if(chargeRate > CHARGE_RATE_MAX)
+    {
+        printf("Charge Rate out of range!\n");
+        return TRUE;
+    }
+    else
+        return FALSE;
+}
+
+float batteryIsOk(float temperature, float soc, float chargeRate) {
+  readtemperature(temperature);
```

Comment:
these parameter check methods are just called and the return values are ignored!! is it by design or a bug?